app/test/vesting_test.go

```diff
@@ -0,0 +1,632 @@
+	for i := range vAccounts {
+		gAccounts = append(gAccounts, authtypes.GenesisAccount(vAccounts[i]))
+	}
+
+	return func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		gs, err := AddGenesisAccountsWithBalancesToGenesisState(gs, gAccounts, balances)
+		assert.NoError(s.T(), err)
+		return gs
+	}
+}
+
+// initPeriodicVestingAccounts function initializes periodic vesting accounts for testing purposes.
+// It takes the count of accounts as input and returns a GenesisOption. It generates account names,
+// base accounts, and balances. It defines vesting periods and creates vesting accounts based on the
+// generated data. The function then updates the genesis state with the generated vesting accounts
+// and balances. The startTime of each account increases progressively to ensure some accounts have
+// locked balances, catering to the testing requirements.
+func (s *VestingModuleTestSuite) initPeriodicVestingAccounts(count int) testnode.GenesisOption {
+	initCoinsForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(initBalanceForGasFee))
+
+	names := testfactory.GenerateAccounts(count)
+	s.accounts.Store(PeriodicVestingAccountType, accountDispenser{names: names})
+	vAccounts, balances := testfactory.GeneratePeriodicVestingAccounts(s.kr, names, initCoinsForGasFee)
+	gAccounts := authtypes.GenesisAccounts{}
+	for i := range vAccounts {
+		gAccounts = append(gAccounts, authtypes.GenesisAccount(vAccounts[i]))
+	}
+
+	return func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		gs, err := AddGenesisAccountsWithBalancesToGenesisState(gs, gAccounts, balances)
+		assert.NoError(s.T(), err)
+		return gs
+	}
+}
+
+// initContinuousVestingAccounts function initializes continuous vesting accounts for testing purposes.
+// It takes the count of accounts as input and returns a GenesisOption. It generates account names,
+// base accounts, and balances. It defines start & end times to creates vesting accounts. The function
+// then updates the genesis state with the generated vesting accounts and balances. The start & endTime
+// of each account increases progressively to ensure some accounts have locked balances, catering to the
+// testing requirements.
+func (s *VestingModuleTestSuite) initContinuousVestingAccounts(count int) testnode.GenesisOption {
+	initCoinsForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(initBalanceForGasFee))
+	names := testfactory.GenerateAccounts(count)
+	s.accounts.Store(ContinuousVestingAccountType, accountDispenser{names: names})
+
+	vAccounts, balances := testfactory.GenerateContinuousVestingAccounts(s.kr, names, initCoinsForGasFee)
+	gAccounts := authtypes.GenesisAccounts{}
+	for i := range vAccounts {
+		gAccounts = append(gAccounts, authtypes.GenesisAccount(vAccounts[i]))
+	}
+
+	return func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		gs, err := AddGenesisAccountsWithBalancesToGenesisState(gs, gAccounts, balances)
+		assert.NoError(s.T(), err)
+		return gs
+	}
+}
```

Comment:
one of the main things that we want from this test and our vesting module efforts is the ability to generically add vesting module accounts to a genesis. Meaning we need to be able to do that outside of this test. The main goal here is to have functionality that add vesting accounts to a genesis, and we can then test that they were installed correctly

Reply:
the changes applied